File: Spa.API/Spa.API/Entities/DichVu.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Spa.API.Entities
{
    [Table("DichVu", Schema = "dbo")]
    public class DichVu
    {
        [Key]
        public int Id { get; set; }

        // FK danh mục (nullable vì SQL đang cho NULL)
        public int? DanhMucId { get; set; }
        public DanhMucDichVu? DanhMucDichVu { get; set; }


        [Required, MaxLength(50)]
        public string MaDichVu { get; set; } = null!;

        [Required, MaxLength(200)]
        public string TenDichVu { get; set; } = null!;

        [MaxLength(500)]
        public string? MoTaNgan { get; set; }

        public string? MoTa { get; set; }          // NVARCHAR(MAX)
        public string? LieuTrinh { get; set; }     // NVARCHAR(MAX)

        [MaxLength(1000)]
        public string? GhiChuNoiBo { get; set; }

        // Giá
        [Column(TypeName = "decimal(18,2)")]
        public decimal GiaHienTai { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? GiaGoc { get; set; }

        [Column(TypeName = "decimal(5,2)")]
        public decimal? PhanTramGiam { get; set; }

        public DateTime? ApDungTu { get; set; }
        public DateTime? ApDungDen { get; set; }

        // Thời lượng / gói
        public int ThoiLuongPhut { get; set; }
        public int? SoBuoi { get; set; }
        public int? ThoiGianHieuLucNgay { get; set; }

        // Bán hàng / vận hành
        [MaxLength(50)]
        public string? SKU { get; set; }

        [MaxLength(50)]
        public string? Barcode { get; set; }

        public bool IsCombo { get; set; }
        public bool IsOnlineVisible { get; set; }
        public bool IsActive { get; set; }
        public int ThuTuHienThi { get; set; }

        // Hình ảnh
        [MaxLength(500)]
        public string? HinhAnhChinh { get; set; }

        public string? HinhAnhJson { get; set; }   // NVARCHAR(MAX) chứa JSON

        // SEO
        [MaxLength(250)]
        public string? Slug { get; set; }

        [MaxLength(250)]
        public string? MetaTitle { get; set; }

        [MaxLength(500)]
        public string? MetaDescription { get; set; }

        // Audit
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        [MaxLength(128)]
        public string? CreatedBy { get; set; }

        [MaxLength(128)]
        public string? UpdatedBy { get; set; }
    }
}

File: admin-spa/src/api/dichVu.ts
import { api } from "../lib/api";

export type DichVuRow = {
  id: number;
  maDichVu: string;
  tenDichVu: string;
  giaHienTai: number;
  giaGoc: number | null;
  thoiLuongPhut: number | null;
  isActive: boolean;
  isOnlineVisible: boolean;
  danhMucId: number | null;
  danhMucTen: string | null;
};

export type DichVuDetail = {
  id: number;
  maDichVu: string;
  tenDichVu: string;
  moTaNgan: string | null;
  moTa: string | null;
  lieuTrinh: string | null;
  giaHienTai: number;
  giaGoc: number | null;
  phanTramGiam: number | null;
  apDungTu: string | null;
  apDungDen: string | null;
  thoiLuongPhut: number | null;
  soBuoi: number | null;
  thoiGianHieuLucNgay: number | null;
  sku: string | null;
  barcode: string | null;
  isCombo: boolean;
  isOnlineVisible: boolean;
  isActive: boolean;
  thuTuHienThi: number | null;
  hinhAnhChinh: string | null;
  hinhAnhJson: string | null;
  slug: string | null;
  metaTitle: string | null;
  metaDescription: string | null;
  danhMucId: number | null;
  danhMucTen: string | null;
};

export type CreateDto = {
  danhMucId?: number | null;
  maDichVu: string;
  tenDichVu: string;
  moTaNgan?: string | null;
  moTa?: string | null;
  lieuTrinh?: string | null;
  giaHienTai: number;
  giaGoc?: number | null;
  phanTramGiam?: number | null;
  apDungTu?: string | null;
  apDungDen?: string | null;
  thoiLuongPhut?: number | null;
  soBuoi?: number | null;
  thoiGianHieuLucNgay?: number | null;
  sku?: string | null;
  barcode?: string | null;
  isCombo?: boolean;
  isOnlineVisible?: boolean;
  isActive?: boolean;
  thuTuHienThi?: number | null;
  hinhAnhChinh?: string | null;
  hinhAnhJson?: string | null;
  slug?: string | null;
  metaTitle?: string | null;
  metaDescription?: string | null;
};

export type UpdateDto = Omit<CreateDto, "maDichVu"> & {
  tenDichVu: string;
  giaHienTai: number;
};

export const dvApi = {
  list: async (params: { active?: boolean; danhMucId?: number; q?: string }) => {
    const sp = new URLSearchParams();
    if (params.active !== undefined) sp.set("active", String(params.active));
    if (params.danhMucId !== undefined) sp.set("danhMucId", String(params.danhMucId));
    if (params.q) sp.set("q", params.q);
    const url = sp.toString() ? `/api/dich-vu?${sp}` : "/api/dich-vu";
    return (await api.get<DichVuRow[]>(url)).data;
  },
  get: async (id: number) => (await api.get<DichVuDetail>(`/api/dich-vu/${id}`)).data,
  create: async (dto: CreateDto) => (await api.post("/api/dich-vu", dto)).data,
  update: async (id: number, dto: UpdateDto) => (await api.put(`/api/dich-vu/${id}`, dto)).data,
  remove: async (id: number) => (await api.delete(`/api/dich-vu/${id}`)).data,
};
